### DISCUSSION TITLE
How to efficiently decode XML encoded string from Span<char>

### DISCUSSION THREAD

**Discussion by lsoft** [dotnet/runtime#81328] [Q&A] [answered]
Hi!

I have a XML encoded string in `Span<char>`, and I need to get decoded `string`. I have the following solution now:

```C#
var mystring = @"my string !@#$%^&amp;*()_+|-=\&#39;;[]{},./&lt;&gt;?".AsSpan();
string result = global::System.Net.WebUtility.HtmlDecode(mystring.ToString());
return result;
```

I would to get rid of `ToString` for intermediate string, but I failed to find such alternative in .NET. Any approaches or ideas, guys?

Thanks!

**Comment by huoyaoyuan** [comment, marked as answer]
There isn't span related API for xml yet. Xml was not focused for optimization in these years.

**Reply by lsoft** [reply]
@huoyaoyuan thanks for info. I am wondering: will it be helpful for tracking purposes to convert this discussion into the issue?

**Reply by huoyaoyuan** [reply]
I think it's better to open a new issue, and specify the API and scope you want to get improved.

**Reply by lsoft** [reply]
@huoyaoyuan thanks! https://github.com/dotnet/runtime/issues/81411
